api/src/pcapi/scripts/beneficiary_fraud_check/remove_dms_content_tzinfo.py

```diff
@@ -0,0 +1,93 @@
+from pcapi.models import db
+from pcapi.repository import repository
+
+
+logging.basicConfig()
+logger = logging.getLogger("my-logger")
+logger.setLevel(logging.DEBUG)
+
+
+def timed(func):  # type: ignore[no-untyped-def]
+    """This decorator prints the execution time for the decorated function."""
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):  # type: ignore[no-untyped-def]
+        start = time.time()
+        result = func(*args, **kwargs)
+        end = time.time()
+        logger.debug("%s ran in %ss", func.__name__, round(end - start, 2))
+        return result
+
+    return wrapper
+
+
+def clear_line(n: int = 1) -> None:
+    LINE_UP = "\033[1A"
+    LINE_CLEAR = "\x1b[2K"
+    for _ in range(n):
+        print(LINE_UP, end=LINE_CLEAR)
+
+
+def display_progress(start_time: float, current: int, total: int) -> None:
+    progression = current / total
+    elapsed_time = time.time() - start_time
+    duration = elapsed_time / progression
+    timestamp_eta = start_time + duration
+    datetime_eta = datetime.datetime.fromtimestamp(timestamp_eta).astimezone(pytz.timezone("Europe/Paris"))
+    time_eta = datetime.datetime.strftime(datetime_eta, "%H:%M:%S")
+    print(f"{current:6d} / {total} -> {progression * 100:3.2f}% in {elapsed_time:.2f}s")
+    print(f"ETA {time_eta} Estimated duration {duration:.2f}s")
+    clear_line()
+
+
+@timed
+def remove_dms_content_tzinfo(dry_run: bool = True, batch_size: int = 1000) -> None:
+    query = fraud_models.BeneficiaryFraudCheck.query.filter(
+        fraud_models.BeneficiaryFraudCheck.type == fraud_models.FraudCheckType.DMS,
+        fraud_models.BeneficiaryFraudCheck.resultContent.isnot(None),
+        fraud_models.BeneficiaryFraudCheck.resultContent["registration_datetime"].isnot(None),
+    ).order_by(fraud_models.BeneficiaryFraudCheck.id)
+    total = 196068  # query.count()
+    print(f"{total} beneficiary fraud checks to update")
+
+    start_index = 0
+    start_time = time.time()
+    subquery = query
+    while start_index < total:
+        subquery = query.filter(fraud_models.BeneficiaryFraudCheck.id > start_index)
+        bfcs: list[fraud_models.BeneficiaryFraudCheck] = subquery.limit(batch_size).all()
+        for bfc in bfcs:
+            if not bfc.resultContent:
+                continue
+
+            old = datetime.datetime.fromisoformat(bfc.resultContent["registration_datetime"])
+            if not old.tzinfo:
+                continue
+
+            bfc.resultContent["registration_datetime"] = parse_dms_datetime(old)
+            flag_modified(bfc, "resultContent")
+
+        if dry_run:
+            db.session.rollback()
+        else:
+            repository.save(*bfcs)
+            assert all(
+                not datetime.datetime.fromisoformat(bfc.resultContent["registration_datetime"]).tzinfo
+                for bfc in bfcs
+                if bfc.resultContent and bfc.resultContent.get("registration_datetime")
+            )
```

Comment:
Est-ce qu'il ne vaut mieux pas appeler `.save()` que si la vérification passe ? Et un _rollback_ sinon.

Reply:
Justement non : j'ai commencé à mettre un `assert` après avoir eu des bugs dans la sauvegarde de mes objets en BDD. C'est pour ça que je ne le fait qu'en cas de `dry_run is False`. Depuis, ces bugs sont corrigés mais j'ai quand même laisser le `assert`.